server/controllers/notifications.js

```diff
@@ -0,0 +1,76 @@
+const Notification = require("../models/Notifications");
+const asyncHandler = require("express-async-handler");
+const mongoose = require('mongoose');
+const { ObjectId } = mongoose.Types;
+
+// @route POST /notification
+// @desc create a new notification
+// @access Private
+exports.createNotification = asyncHandler(async (req, res, next) => {
+  try {
+    const userId = req.user.id;
+    
+    const { type, date, title, description } = req.body;
+    const notificatonModel = {
+        userId: ObjectId(userId),
+        date: new Date(date),
+        type,
+        title,
+        description,
+    };
+    
+    const notification = await Notification.create(notificatonModel); 
+    res.status(201).json({
+      success: { notification }
+    });
+  } catch (error) {
+    next(error);
+  } 
+});
+
+// @route GET /notification/all
+// @desc list of notifications; only get read if query provided 
+// @access Private
+exports.notificationList = asyncHandler(async (req, res, next) => {
+  try {
+    const userId = req.user.id;
+    const { unread } = req.query;
```

Comment:
Query parameters have not been defined as part of the route, please add them 

Also, if unread === 'true', shouldn't we returning the notifications which haven't bee read? (read: false)

Reply:
@rajivtitus how do we define optional/queries? I thought only params were defined in the route: 
notification/:notificationId => notification/92i391i3ej32
notification/ => notification/?unread=true

Reply:
> Also, if unread === 'true', shouldn't we returning the notifications which haven't been read? (read: false)
> 
yes I changed the prop to false.

Reply:
> Yes I have changed thAlso, if unread === 'true', shouldn't we returning the notifications which haven't been read? (read: false)
true, I have changed the prop to false

Reply:
@iamalx My apologies, please ignore the comment about query parameters (I must've lost my train of thought) 😄